erc1155/src/lib.rs

```diff
@@ -185,56 +187,85 @@ impl ExtendERC1155TokenBase for ERC1155Token {
                 return false;
             }
         }
-
         true
     }
 
-    fn mint(&mut self, account: &ActorId, id: &u128, amount: u128) {
+    fn mint(&mut self, account: &ActorId, id: &u128, amount: u128, meta: Option<TokenMetadata>) {
         if account == &ZERO_ID {
             panic!("ERC1155: Mint to zero address")
         }
+        if let Some(metadata) = meta {
+            // Off you go...
+            if amount > 1 {
+                panic!("ERC1155: Mint metadata to a fungible token")
+            }
+            self.token_metadata.insert(*id, metadata);
+        }
         let prev_balance = self.balance_of(account, id);
         self.set_balance(account, id, prev_balance.saturating_add(amount));
     }
 
-    fn mint_batch(&mut self, account: &ActorId, ids: &[u128], amounts: &[u128]) {
+    fn mint_batch(
+        &mut self,
+        account: &ActorId,
+        ids: &[u128],
+        amounts: &[u128],
+        meta: &[Option<TokenMetadata>],
```

Comment:
I don't see the reason to have a slice here, then clone contents instead of moving.

```suggestion
        meta: Vec<Option<TokenMetadata>>,
```